src/Admin/AbstractAdmin.php

```diff
@@ -1302,7 +1302,7 @@ final public function getChild(string $code): AdminInterface
         return $this->getChildren()[$code];
     }
 
-    final public function setParent(AdminInterface $parent, string $parentAssociationMapping): void
+    final public function setParent(AdminInterface $parent, ?string $parentAssociationMapping): void
```

Comment:
should we avoid call parent?

Something like 

if $parent !== null $child->setParent()

Reply:
Set parent is doing
```
final public function setParent(AdminInterface $parent, ?string $parentAssociationMapping): void
    {
        $this->parent = $parent;
        $this->parentAssociationMapping[$parent->getCode()] = $parentAssociationMapping;
    }
```

So, we still need to 
- set the parent
- fill the parentAssociationMapping because later we acces it with
```
final public function getParentAssociationMapping(): ?string
    {
        if (!$this->isChild()) {
            throw new \LogicException(sprintf(
                'Admin "%s" has no parent.',
                static::class
            ));
        }

        $parent = $this->getParent()->getCode();

        return $this->parentAssociationMapping[$parent];
    }
```
which already accepts to return null.